apps/enc.c

```diff
@@ -347,6 +360,31 @@ int enc_main(int argc, char **argv)
         }
     }
 
+    if (cipher && (EVP_CIPHER_flags(cipher) & EVP_CIPH_FLAG_AEAD_CIPHER) && (aad == NULL)) {
+#ifndef OPENSSL_NO_UI
+        for (;;) {
+            char prompt[200];
+
+            BIO_snprintf(prompt, sizeof prompt, "enter %s %s aad:",
+                         OBJ_nid2ln(EVP_CIPHER_nid(cipher)),
+                         (enc) ? "encryption" : "decryption");
+            aadbuf[0] = '\0';
+            i = EVP_read_aad_string((char *)aadbuf, SIZE, prompt);
+            if (i == 0) {
```

Comment:
So EVP_read_aad_string() returns 0 on success? That is contrary to our normal way of doing things (1 is success, 0 (and sometimes -1) is error). Mind you the UI APIs are slightly different so I defer to @levitte on the right way to do this.